File: src/ua/com/sevenpowerx/kotlin/book/kotlin/glava_4/java_class/Button.java
package ua.com.sevenpowerx.kotlin.book.kotlin.glava_4.java_class;

import ua.com.sevenpowerx.kotlin.book.kotlin.glava_4.use_interface.State;
import ua.com.sevenpowerx.kotlin.book.kotlin.glava_4.use_interface.View;

public class Button implements View {
	@Override
	public State getCurrentState() {
		return new ButtonState();
	}
	
	@Override
	public void restoreState(State state) {
	/*...*/
	}
	
	static private class ButtonState implements State {
	}
}

File: src/ua/com/sevenpowerx/kotlin/book/kotlin/glava_4/kotlin_class/Button.kt
package ua.com.sevenpowerx.kotlin.book.kotlin.glava_4.kotlin_class

import ua.com.sevenpowerx.kotlin.book.kotlin.glava_4.use_interface.State
import ua.com.sevenpowerx.kotlin.book.kotlin.glava_4.use_interface.View

class Button: View
{
	override fun getCurrentState(): State = ButtonState()
	override fun restoreState(state: State) {/*...*/	}

	inner class ButtonState : State {/*...*/}

}